services/QuillLMS/client/app/bundles/Teacher/components/accounts/edit/teacher_email_notifications.jsx

```diff
@@ -1,36 +1,172 @@
 import React from 'react';
+
+import { DropdownInput, informationIcon, } from '../../../../Shared/index';
+
 const smallWhiteCheckSrc = `${process.env.CDN_URL}/images/shared/check-small-white.svg`
 
-export default class TeacherEmailNotifications extends React.Component {
-  toggleSendNewsletter = () => {
-    const { sendNewsletter, updateUser, } = this.props
-    const data = {
-      send_newsletter: !sendNewsletter,
-      school_options_do_not_apply: true
-    };
-    updateUser(data, '/teachers/update_my_account', 'Settings saved')
+const TeacherEmailNotifications = ({ activateSection, active, deactivateSection, passedSendNewsletter, passedNotificationSettings, passedNotificationEmailFrequency, updateUser, updateNotificationSettings, updateTeacherInfo }) => {
+
+  const [sendNewsletter, setSendNewsletter] = React.useState(passedSendNewsletter)
+  const [notificationSettings, setNotificationSettings] = React.useState(passedNotificationSettings)
+  const [notificationEmailFrequency, setNotificationEmailFrequency] = React.useState(passedNotificationEmailFrequency)
+
+  React.useEffect(() => {
+    if (active) { return }
+
+    reset()
+  }, [active])
+
+  // The following three useEffect calls should be unnecesary, but I was finding
+  // in testing that during render cycles where "passed" values updated, the useState
+  // calls weren't updating, so the UI would revert to the original state until refresh.
+  // I suspect that it might be because the save call makes three separate API calls,
+  // but at least adding these hooks makes things work.
```

Comment:
So I think what you're trying to achieve with using passed values for the initial state that will update on re-render can be achieved with passing a key prop to this `TeacherEmailNotifications` component-- this post goes into a bit more detail of how that works: https://tkdodo.eu/blog/putting-props-to-use-state